lblod/ember-rdfa-editor
# Improvements to the UX of links

### Overview
Some UX improvements I wanted to add when working on setting hrefs from urls. When doing that ticket I struggled to make these work, so gave up. I re-visited them to see if there was something missing from ember-nodes, but it turns out I was just misusing them. I also made 2 very minor fixes by tweaking the layout.

In terms of UX, this to me feels much more natural (or at least close to other editors). If we're unsure about any parts of it, we can split either of the 2 UX tweaks or 2 layout fixes by removing some commits.

##### connected issues and PRs:
Issues fixed:
https://binnenland.atlassian.net/browse/GN-4504
https://binnenland.atlassian.net/browse/GN-4480

### Setup
N/A

### How to test/reproduce
See linked issues for reproduction instructions for those. For the UX tweaks:
- Type a valid url or email address as normal text, select it, then click the 'insert link' button on the toolbar. The link should use the url (with added http:// or mailto:) as the href, instead of requiring to manually enter both the text and the href.
- With no selection, click the insert link button, then enter a valid url or email address (without changing the selection). This will set this to be the text content of the link. Then click somewhere else to de-select the link. This will automatically set the href to be the one from the text content.

### Challenges/uncertainties
N/A

### Checks PR readiness
